src/SFA.DAS.Assessor.Functions/Domain/Print/Services/BlobFileTransferClient.cs

```diff
@@ -224,11 +223,36 @@ private async Task<List<CloudBlob>> GetBlobsHierarchicalListingAsync(CloudBlobDi
             }
             catch (StorageException ex)
             {
-                _logger.LogError(ex, $"Error listing file in blob storage {ContainerName}");
+                _logger.LogError(ex, $"Error listing file in blob storage {_containerName}");
                 throw;
             }
 
             return blobs;
         }
+
+        public string GetContainerSasUri(string ipAddress, DateTime expiryTime, SharedAccessBlobPermissions? permissions = null)
+        {
+            var account = CloudStorageAccount.Parse(_connectionString);
+            var client = account.CreateCloudBlobClient();
+            var container = client.GetContainerReference(_containerName);
+
+            SharedAccessBlobPolicy adHocPolicy = new SharedAccessBlobPolicy()
+            {
+                SharedAccessExpiryTime = expiryTime,
+                Permissions = permissions ??
+                    SharedAccessBlobPermissions.Read |
+                    SharedAccessBlobPermissions.Write |
+                    SharedAccessBlobPermissions.Create |
+                    SharedAccessBlobPermissions.List |
+                    SharedAccessBlobPermissions.Delete
+            };
+
+            var ipAddressOrRange = !string.IsNullOrEmpty(ipAddress)
```

Comment:
Could this raise an error or warning if the IP Address is NullOrEmpty and not generate a token or send email

Reply:
Changes look good to me 👍